man/systemd.network.xml

```diff
@@ -2257,6 +2257,53 @@
       </variablelist>
   </refsect1>
 
+  <refsect1>
+    <title>[TrafficControlQueueingDiscipline] Section Options</title>
+    <para>The <literal>[TrafficControlQueueingDiscipline]</literal> section manages the Traffic control. It can be used
+    to configure the kernel packet scheduler and simulate packet delay and loss for UDP or TCP applications,
+    or limit the bandwidth usage of a particular service to simulate internet connections.</para>
+
+    <variablelist class='network-directives'>
+      <varlistentry>
+        <term><varname>Parent=</varname></term>
+        <listitem>
+          <para>Specifies Queueing Discipline (qdisc). Takes one of <literal>root</literal> or <literal>clsact</literal>.
+          Defaults to <literal>root</literal>.</para>
+        </listitem>
+      </varlistentry>
+
+      <varlistentry>
+        <term><varname>NetworkDelaySec=</varname></term>
+        <listitem>
+          <para>Specifies the network fixed amount of delay to all packets going out of the interface. Defaults to unset.</para>
+        </listitem>
+      </varlistentry>
+
+      <varlistentry>
+        <term><varname>NetworkDelayJitterSec=</varname></term>
+        <listitem>
+          <para>Specifies the random delay to be added to <varname>NetworkDelaySec=</varname>. Defaults to unset.</para>
+        </listitem>
+      </varlistentry>
+
+      <varlistentry>
+        <term><varname>NetworkDelayLimitPacket=</varname></term>
+        <listitem>
+          <para>Specifies the fifo limit (packets). An unsigned integer ranges 0 to 4294967294. Defaults to 100.</para>
+        </listitem>
+      </varlistentry>
+
+      <varlistentry>
+        <term><varname>NetworkDelayLossQuota=</varname></term>
+        <listitem>
+          <para>Specifies the random packet loss in percent to be added to <varname>NetworkDelaySec=</varname>.
```

Comment:
"to be added to..." doesn't make sense. How do you add a loss percentage to a delay?

Reply:
What do you mean. Can you explain

Reply:
Packet loss and delay are independent from one another. The packet loss is not "added to" the delay. It's just a completely different thing.

Reply:
Loss added to delay means not all packets travels. What is so difficult to understand

Reply:
I don't know how else to explain. Loss and delay are two independent phenomena. Even mathematically, you cannot add a loss (expressed in %) to a delay (expressed in ms). It just doesn't make sense.

Just write "Specifies the random packet loss in percent."

Reply:
Loss and delay How you simulate it ? For example TCP/UDP

Reply:
I don't understand your comment

Reply:
For TCP if you drop the packet it will retransmit but UDP will not .  So how you simulate they drop/loss  in UDP ?

Reply:
With UDP the kernel will not retransmit, as you said. It's the application's responsibility to retransmit a UDP datagram (if it wants to).